crates/mpt/src/node.rs

```diff
@@ -169,33 +141,26 @@ impl TrieNode {
                     }
                 }
             }
-            TrieNode::Leaf { key, value } => {
-                let key_nibbles = Nibbles::unpack(key.clone());
-                let shared_nibbles = key_nibbles[1..].as_ref();
-
+            TrieNode::Leaf { prefix, value } => {
                 // If the key length is one, it only contains the prefix and no shared nibbles.
                 // Return the key and value.
-                if key.len() == 1 || nibble_offset + shared_nibbles.len() >= path.len() {
+                if prefix.len() == 1 || nibble_offset + prefix.len() >= path.len() {
```

Comment:
This seems to change the accounting here where `shared_nibbles` was 1 less than the now `prefix.len()`. Is this correct?

Reply:
Yes great spot - you are correct that this is a bug. This is fixed in #196 @ https://github.com/ethereum-optimism/kona/pull/196/files#diff-f3393c37018f3745aed7a3c1f44de945d0804f997516c18372d692d48c39a13aL145-R145

Reply:
Ah thank you!